app/models/proposal.rb

```diff
@@ -95,18 +89,15 @@ def votes_percentage
   end
 
   def editable?(current_user)
-    return false unless current_user
-    current_user && votes_count < 2 && user_id == current_user.id
+    current_user && votes_count < 2 && current_user == self.user
   end
 
   def current_user_support?(current_user)
-    return false unless current_user
-    cu_votes = votes.count { |v| v.user_id == current_user.id }
-    return cu_votes.present?
+    current_user && self.votes.find_by(user_id: current_user.id).present?
   end
 
   def recent_vote
     return self.created_at if self.votes.blank?
```

Comment:
Mid-method returns aren't my favorite, but this is probably the best way to do this without doing something hard-to-read like `(self.votes.by_recency.first || self).created_at`